cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -839,6 +848,8 @@ public String getOutputFormatClassName() {
       outputArgs.put(Constants.Dataset.Partitioned.HCONF_ATTR_OUTPUT_FORMAT_CLASS_NAME,
                      files.getOutputFormatClassName());
       outputArgs.put(Constants.Dataset.Partitioned.HCONF_ATTR_OUTPUT_DATASET, getName());
+    } else {
+      assertNotExists(outputKey);
```

Comment:
This method is probably only called by CDAP's mapreduce/spark runtime. Is that inside a transaction?